azurerm/internal/services/mysql/tests/mysql_server_data_source_test.go

```diff
@@ -0,0 +1,205 @@
+	data := acceptance.BuildTestData(t, "data.azurerm_mysql_server", "test")
+	mysqlVersion := "5.7"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMMySQLServerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAzureRMMySQLServer_autogrow(data, mysqlVersion),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMySQLServerExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "sku_name", "GP_Gen5_2"),
+					resource.TestCheckResourceAttr(data.ResourceName, "administrator_login", "acctestun"),
+					resource.TestCheckResourceAttr(data.ResourceName, "auto_grow_enabled", "true"),
+					resource.TestCheckResourceAttr(data.ResourceName, "ssl_minimal_tls_version_enforced", "TLS1_1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "storage_mb", "51200"),
+					resource.TestCheckResourceAttr(data.ResourceName, "version", "5.7"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAzureRMMySQLServer_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_mysql_server", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMMySQLServerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAzureRMMySQLServer_complete(data, "8.0"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMySQLServerExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "sku_name", "GP_Gen5_2"),
+					resource.TestCheckResourceAttr(data.ResourceName, "administrator_login", "acctestun"),
+					resource.TestCheckResourceAttr(data.ResourceName, "auto_grow_enabled", "false"),
+					resource.TestCheckResourceAttr(data.ResourceName, "ssl_minimal_tls_version_enforced", "TLS1_1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "storage_mb", "51200"),
+					resource.TestCheckResourceAttr(data.ResourceName, "version", "8.0"),
+					resource.TestCheckResourceAttr(data.ResourceName, "threat_detection_policy.#", "1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "threat_detection_policy.0.enabled", "true"),
+					resource.TestCheckResourceAttr(data.ResourceName, "threat_detection_policy.0.email_account_admins", "true"),
+					resource.TestCheckResourceAttr(data.ResourceName, "threat_detection_policy.0.retention_days", "7"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAzureRMMySQLServer_basic(data acceptance.TestData, version string) string {
+	return fmt.Sprintf(`
+%s
+
+data "azurerm_mysql_server" "test" {
+  name                = "acctestmysqlsvr-%d"
+  resource_group_name = azurerm_resource_group.test.name
+}
+`, testAccAzureRMMySQLServer_basic(data, version), data.RandomInteger)
```

Comment:
The data source needs to reference an original resource attribute so Terraform can infer the correct order, as without it, it tries to look up the MySQL server before it's created.

```suggestion
  name                = azurerm_mysql_server.test.name
  resource_group_name = azurerm_resource_group.test.name
}
`, testAccAzureRMMySQLServer_basic(data, version))
```